### PR TITLE
fix doc

### PR DESCRIPTION
**Pull request by FuGangqiang**

### CODE REVIEW

File: src/doc/trpl/generics.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ fn takes_anything<T>(x: T) {
 ```
 
 The syntax has two parts: the `<T>` says “this function is generic over one
-type, `T`”, and the `x: T` says “x has the type `T`.”
+type, `T`”, and the `x: T` says “x has the type `T`”.
```
Comment (steveklabnik) [inline src/doc/trpl/generics.md]: commas go inside of double quotes, actually.

File: src/doc/trpl/enums.md [hunk 2]
```diff
@@ -28,6 +28,16 @@ In `Character`, for instance, `Digit` gives a meaningful name for an `i32`
...
+We use the `::` syntax to use the name of each variant: They’re scoped by the name
+of the `enum` itself. This allows both of these to work:
+
+```rust,ignore
+Character::Digit(10);
+Hand::Digit;
+```
+
+Both variants are named `Digit`, but they’re scoped to the `enum` name.
+
```
Comment (steveklabnik) [inline src/doc/trpl/enums.md]: What's the reason for this?

Comment (FuGangqiang) [inline src/doc/trpl/enums.md]: I think  these lines should be put here rather than before.
the before splite the explainations of ` equality operators`.

File: src/doc/trpl/generics.md (hunk 1 above)
Comment (FuGangqiang) [inline src/doc/trpl/generics.md]: sorry for that, my english is not very good,maybe I'm wrong.

File: src/doc/trpl/enums.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 % Enums
 
-Rust has a ‘sum type’, an `enum`. Enums are an incredibly useful feature of
+Rust has a‘sum type’, an `enum`. Enums are an incredibly useful feature of
```
Comment (pnkfelix) [inline src/doc/trpl/enums.md]: why did you delete the space between `a` and `'sum type'` here?

File: src/doc/trpl/macros.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ or cumbersome to express that pattern as a generic function, a trait, or
 anything else within Rust’s semantics.
 
 Macros allow us to abstract at a syntactic level. A macro invocation is
-shorthand for an "expanded" syntactic form. This expansion happens early in
+shorthand for an‘expanded’syntactic form. This expansion happens early in
```
Comment (pnkfelix) [inline src/doc/trpl/macros.md]: again, is it okay to delete the space after `an` here?

File: src/doc/trpl/enums.md (hunk 3 above)
Comment (FuGangqiang) [inline src/doc/trpl/enums.md]: en,they shouldn't be deleted, I will turn it back!

**Comment by FuGangqiang** [comment]
the doc has been fixed
